### PR TITLE
Basic chaos endpoint

### PR DESCRIPTION
**Pull request by Sivarion**

### CODE REVIEW

File: envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt [hunk 1]
```diff
@@ -61,6 +68,11 @@ class SnapshotDebugController(controlPlane: ControlPlane) {
         }
     }
 
+    @PostMapping("/chaos/fault/read_network_delay")
```
Comment (pbetkier) [inline envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt:71]: how about `read-network-delay`, according to the convention we already have, e.g. `snapshot-global`

File: envoy-control-runner/src/main/resources/application-docker.yaml [hunk 2]
```diff
@@ -2,4 +2,8 @@ envoy-control:
   source:
     consul:
       host: consul
-      port: 8500
\ No newline at end of file
+      port: 8500
+
+chaos:
+  username: "user"
+  password: "{noop}pass"
```
Comment (pbetkier) [inline envoy-control-runner/src/main/resources/application-docker.yaml:9]: what is a `{noop}`?

File: envoy-control-runner/src/main/resources/application-docker.yaml [hunk 3]
```diff
@@ -2,4 +2,8 @@ envoy-control:
   source:
     consul:
       host: consul
-      port: 8500
\ No newline at end of file
+      port: 8500
+
+chaos:
```
Comment (pbetkier) [inline envoy-control-runner/src/main/resources/application-docker.yaml:7]: I have a feeling you will soon have to migrate these properties to a deeper properties tree, but it's an incubating feature so it's fine.

File: envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt [hunk 4]
```diff
@@ -61,6 +68,11 @@ class SnapshotDebugController(controlPlane: ControlPlane) {
...
+    fun readNetworkDelay(): HttpStatus {
```
Comment (pbetkier) [inline envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt:72]: chaos has nothing to do with snapshots, so move this to its own controller

File: envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt [hunk 5]
```diff
@@ -109,6 +121,36 @@ class SnapshotDebugController(controlPlane: ControlPlane) {
         }
     }
 
+    @Configuration
+    class SecurityConfig : WebSecurityConfigurerAdapter() {
```
Comment (pbetkier) [inline envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt:125]: This also shouldn't be in a snapshot controller.

File: envoy-control-runner/src/main/resources/application-docker.yaml (hunk 2 above)
Comment (Sivarion) [reply envoy-control-runner/src/main/resources/application-docker.yaml:9]: {noop} idicates password encryption - in this case plain text will be used.

File: envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt (hunk 1 above)
Comment (Sivarion) [reply envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugController.kt:71]: Sure, no problem :)

**Pull request merged**